desktopexporter/factory.go

```diff
@@ -17,20 +17,23 @@ const (
 )
 
 // Creates a factory for the Desktop Exporter
-func NewFactory() exporter.Factory {
+func NewFactory(dbFilename string) exporter.Factory {
```

Comment:
Not qualified to do a full review here. but I think there is no need to add dbFilename as argument here and pass it to createDefaultConfig.
If you add a line in main.go as part of `set.ConfigProviderSettings.ResolverSettings.URIs` you can set a the value for the db filename using `yaml:exporters::desktop::dbfilename: ` + dbFilenameFlag

Can scope the flag variable to the `newCommand` and have mapstructure populate the value based on the flag.

This way you can skip the value passing and have the value populated based on the flag.